plutus-tutorial/tutorial/Tutorial/02-validator-scripts.md

```diff
@@ -0,0 +1,205 @@
+
+We also need the `{-# OPTIONS_GHC -O0 #-}` compiler option. It disables some of GHC's optimisations to ensure that the generated Haskell Core can be translated to Plutus IR easily.
+
+```haskell
+import qualified Language.PlutusTx            as P
+import qualified Ledger.Interval              as P
+import           Ledger                       (Address, DataScript(..), PubKey(..), RedeemerScript(..), Signature(..), Slot(..), TxId, ValidatorScript(..))
+import qualified Ledger                       as L
+import qualified Ledger.Ada                   as Ada
+import           Ledger.Ada                   (Ada)
+import           Ledger.Validation            (PendingTx(..), PendingTxIn(..), PendingTxOut)
+import qualified Ledger.Validation            as V
+import           Wallet                       (WalletAPI(..), WalletDiagnostics(..), MonadWallet, EventHandler(..), EventTrigger)
+import qualified Wallet                       as W
+import           Prelude                      hiding ((&&))
+import           GHC.Generics                 (Generic)
+
+import qualified Data.ByteString.Lazy.Char8   as C
+```
+
+The module [`Ledger.Validation`](https://input-output-hk.github.io/plutus/wallet-api-0.1.0.0/html/Ledger-Validation.html), imported as `V`, contains types and functions that can be used in on-chain code. `Language.PlutusTx` lets us translate code between Haskell and Plutus Core (see the [PlutusTx tutorial](./01.plutus-tx.md)). [`Ledger`](https://input-output-hk.github.io/plutus/wallet-api-0.1.0.0/html/Ledger.html) has data types for the ledger model and [`Wallet`](https://input-output-hk.github.io/plutus/wallet-api-0.1.0.0/html/Wallet.html) is the wallet API. [`Wallet.Emulator`](https://input-output-hk.github.io/plutus/wallet-api-0.1.0.0/html/Wallet-Emulator.html) covers interactions with the wallet, for example generating the transactions that actually get the crowdfunding contract onto the blockchain.
```

Comment:
The description of `Wallet.Emulator` seems weird - shouldn't it be about the emulator?

Reply:
Yeah. The module isn't even imported anymore so I just deleted the reference to `Wallet.Emulator` and kept the rest.